src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -762,6 +788,102 @@ where pattern.IsMatch(alias)
         }
 
+        private static List<(string name, string description)> GetCommandDescriptions(string wordToComplete)
+        {
+            var context = LocalPipeline.GetExecutionContextFromTLS();
+            var helpRequest = new HelpRequest(wordToComplete, HelpCategory.Cmdlet | HelpCategory.Function)
+            {
+                ProviderContext = new ProviderContext(null, context, context.SessionState.Path)
+            };
+            var helpInfos = context.HelpSystem.GetHelp(helpRequest);
+            if (helpInfos == null)
+                return null;
+            var results = new List<(string name, string description)>(50);
+
+            foreach (var helpInfo in helpInfos)
+            {
+                results.Add((helpInfo.Name, PropertyHelpPSObjectToToolTip(helpInfo.ShortHelp?.Properties["description"]?.Value as PSObject[])));
+            }
+
+            return results;
+        }
+
+        private static List<(string name, string description)> GetParameterDescriptions(string commandName)
+        {
+            var context = LocalPipeline.GetExecutionContextFromTLS();
+            var helpRequest = new HelpRequest(commandName, HelpCategory.Cmdlet | HelpCategory.Function)
+            {
+                ProviderContext = new ProviderContext(null, context, context.SessionState.Path)
+            };
+            var helpInfo = context.HelpSystem.GetHelp(helpRequest).FirstOrDefault();
+            PSObject[] propertyHelpInfos = helpInfo?.GetParameter("*");
+            var propertyDescriptions = new List<(string name, string description)>();
+            foreach (var p in propertyHelpInfos)
+            {
+                var props = p.Properties;
+                var name = PSObject.Base(props["name"]?.Value) as string;
+                var descriptions = PSObject.Base(props["description"]?.Value) as PSObject[];
+
+                var desc = PropertyHelpPSObjectToToolTip(descriptions as PSObject[]);
+                if (name != null && desc != null)
+                    propertyDescriptions.Add((name, desc));
+            }
+
+            return propertyDescriptions;
+        }
+
+        static string PropertyHelpPSObjectToToolTip(PSObject[] description)
+        {
+            if (description == null || description.Length == 0) return "";
+            var text = description[0]?.Properties["Text"]?.Value as string;
+            var builder = new StringBuilder(text.Length + text.Length / 50);
+            int remaining = text.Length;
+            var chars = text.ToCharArray();
+            int begin = 0;
+            int NextLinebreak(int i, int above)
+            {
+                above = above + i;
+                i = text.IndexOf(' ', i);
+                var previous = i;
+                while (i != -1 && i < above)
+                {
+                    previous = i;
+                    i = text.IndexOf(' ', i + 1);
+                }
+                if (i == -1)
+                    return previous;
+
+
```

Comment:
Please remove extra line and use `{...}`.

Reply:
>Braces usually go on their own lines, with the exception of single line statements that are properly indented.